Sources/Vapor/Utilities/Storage.swift

```diff
@@ -34,10 +44,26 @@ public struct Storage {
         }
     }
 
+    /// Read access to a value via keyed subscript, adding the provided default
+    /// value to the storage if the key does not already exist. Similar to
+    /// ``Swift/Dictionary/subscript(key:default:)``. The `defaultValue` autoclosure
+    /// is evaluated only when the key does not already exist in the container.
+    public subscript<Key>(_ key: Key.Type, default defaultValue: @autoclosure () -> Key.Value) -> Key.Value
+        where Key: StorageKey
+    {
+        mutating get {
+            if let existing = self[key] { return existing }
+            self.set(Key.self, to: defaultValue())
+            return self[key]!
```

Comment:
I mean I'm not enamored with the force unwrap even though it's safe. One option would be

```suggestion
            let new = defaultValue()
            self.set(Key.self, to: new)
            return new
```